File: backend/app/models/deal.py
"""
항공 특가·이벤트 응답 모델 (API)
"""
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel


class DealResponse(BaseModel):
    """특가 이벤트 한 건 (앱 푸시용)"""

    id: str
    airline: str
    airline_id: str
    title: str
    description: str | None = None
    url: str
    image_url: str | None = None
    event_start: datetime | None = None
    event_end: datetime | None = None
    routes: list | dict | None = None
    price: Decimal | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: android/app/src/main/java/com/aerofinder/data/Deal.kt
package com.aerofinder.data

import com.google.gson.annotations.SerializedName

/**
 * 백엔드 GET /api/deals 응답 한 건
 */
data class Deal(
    val id: String,
    val airline: String,
    @SerializedName("airline_id") val airlineId: String,
    val title: String,
    val description: String? = null,
    val url: String,
    @SerializedName("image_url") val imageUrl: String? = null,
    @SerializedName("event_start") val eventStart: String? = null,
    @SerializedName("event_end") val eventEnd: String? = null,
    val routes: Any? = null,
    val price: Number? = null,
    @SerializedName("created_at") val createdAt: String? = null,
)
